### PR TITLE
Detect, warn about, and optionally enforce new 'pair_style' command after 'read_restart'

### PR DESCRIPTION
**Pull request by akohlmey**
## Purpose

Several pair styles in LAMMPS do not support reading their state from binary restart files. In these cases it it required to re-issue the `pair_style` and `pair_coeff` commands **after** the `read_restart` command.
This PR provides an infrastructure to detect this and issue a generic warning  (in `Pair::init()`), but individual pair styles can also implement an error exit (in `Pair*::init_style()`) in case the `did_dummy_restart` variable has not been reset before (in `Pair*::settings()`, which is only called when a `pair_style` command is issued. This hard exit has been implemented for a number of pair styles, that explicitly state in their respective documentation, that the `pair_style` must be re-issued after a restart.

## Author(s)

Axel Kohlmeyer (Temple U)

## Backward Compatibility

This change is fully backward compatible for correct inputs. For incorrect inputs the result is either a generic (one time) warning or an error message and stop. The latter happens only for pair styles that are documented and confirmed to require re-issuing the `pair_style` command and have been adapted as needed.

## Implementation Notes

The detection of whether restarts are supported or not is done via the member variable `Pair::did_dummy_restart`, which is set to 1 in the virtual function `Pair::read_restart()`. If a pair style does support restarts, it will override this function and `did_dummy_restart` will remain at its default value of 0. If it value is 1 inside of `Pair::init()` a general warning is printed. However, individual pair styles can test for it in `Pair*::init_style()` and do a hard exit. `did_dummy_restart` is reset to 0 for all pair styles in either the `Pair` constructor or in `Input::pair_style()`.

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The source code follows the LAMMPS formatting guidelines

## Further Information, Files, and Links

N/A

### CODE REVIEW

**Comment by akohlmey** [comment]
@sjplimp do you still have unanswered concerns about this PR?

File: src/PYTHON/pair_python.cpp [hunk 1]
```diff
@@ -392,6 +392,17 @@ void PairPython::coeff(int narg, char **arg)
 
 /* ---------------------------------------------------------------------- */
 
+void PairPython::init_style()
+{
+  if (did_dummy_restart)
+    error->all(FLERR,"Must specify 'pair_style' command after "
+               "'read_restart' for pair style python");
+
+  neighbor->request(this,instance_me);
```
Comment (rbberger) [inline src/PYTHON/pair_python.cpp]: @akohlmey looks like there's a missing neighbor.h here

Comment (akohlmey) [reply src/PYTHON/pair_python.cpp]: should be addressed in dcf2a33

**Comment by akohlmey** [comment]
will submit a new variant of this after the stable release

**Pull request closed without merging**